repository/user_id_provider.go

```diff
@@ -30,11 +29,15 @@ func (r *Repo) InsertUserIdProvider(ctx context.Context, userIdProvider *object.
 	return r.insertEntity(ctx, userIdProvider)
 }
 
-func (r *Repo) UpdateUserIdProvider(ctx context.Context, userIdProvider *object.UserIdProvider) error {
+func (r *Repo) UpdateUserIdProvider(ctx context.Context, userIdProvider *object.UserIdProvider, updateKey string) error {
+	updateValue := userIdProvider.ProviderName
```

Comment:
I think that the repository is not the best place for this logic.
Can we take it to a higher level?

Reply:
fixed